### PR TITLE
[PLAT-10653] Create react native resource attributes

### PR DESCRIPTION
**Pull request by gingerbenw**
## Goal

Create a resource attributes source for react native

The device.id attribute will be covered in a later ticket

## Testing

Added unit test for each attribute, and reconfigured jest for react native

### CODE REVIEW

File: packages/platforms/react-native/lib/resource-attributes-source/resource-attributes-source.test.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        value: { stringValue: '12345678' }
+      },
+      {
+        key: 'deployment.environment',
+        value: { stringValue: 'test' }
+      },
+      {
+        key: 'device.id',
+        value: { stringValue: 'unknown' }
+      },
```
Comment (gingerbenw) [inline packages/platforms/react-native/lib/resource-attributes-source/resource-attributes-source.test.ts:22]: covered in another ticket

File: packages/platforms/react-native/lib/resource-attributes-source/resource-attributes-source.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    config.appVersion,
+    'bugsnag.performance.reactnative',
+    '__VERSION__'
+  )
+
+  attributes.set('os.type', Platform.select({ android: 'linux', ios: 'darwin', default: 'unknown' }))
+  attributes.set('os.name', Platform.OS)
+  attributes.set('os.version', Platform.Version.toString())
+  attributes.set('service.name', config.appName)
+  attributes.set('device.id', 'unknown')
```
Comment (gingerbenw) [inline packages/platforms/react-native/lib/resource-attributes-source/resource-attributes-source.ts:17]: covered in another ticket

File: packages/platforms/react-native/lib/resource-attributes-source/resource-attributes-source.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  if (config.codeBundleId) {
+    attributes.set('bugsnag.app.code_bundle_id', config.codeBundleId)
+  }
+
+  // @ts-expect-error Platform.constants.Manufacturer exists on android devices
+  attributes.set('device.manufacturer', Platform.select({ android: Platform.constants.Manufacturer, ios: 'Apple', default: 'unknown' }))
+
+  // @ts-expect-error Platform.constants.Serial exists on android devices
+  attributes.set('device.model.identifier', Platform.select({ android: Platform.constants.Model, ios: 'unknown', default: 'unknown' }))
```
Comment (gingerbenw) [inline packages/platforms/react-native/lib/resource-attributes-source/resource-attributes-source.ts:27]: known issue: no way to retrieve model on iOS without crossing the native layer

**Comment by gingerbenw** [comment]
no idea why coverage is saying browser clock has been affected 🤷🏻

File: packages/platforms/react-native/lib/resource-attributes-source/resource-attributes-source.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  attributes.set('device.id', 'unknown')
+
+  if (config.codeBundleId) {
+    attributes.set('bugsnag.app.code_bundle_id', config.codeBundleId)
+  }
+
+  // @ts-expect-error Platform.constants.Manufacturer exists on android devices
+  attributes.set('device.manufacturer', Platform.select({ android: Platform.constants.Manufacturer, ios: 'Apple', default: 'unknown' }))
+
+  // @ts-expect-error Platform.constants.Serial exists on android devices
```
Comment (lemnik) [inline packages/platforms/react-native/lib/resource-attributes-source/resource-attributes-source.ts:26]:
```suggestion
  // @ts-expect-error Platform.constants.Model exists on android devices
```

**Comment by yousif-bugsnag** [comment]
Was there a particular reason for placing tests in the same directory as the source files? IMO as we add more files this will make it harder to navigate in a tree view

**Comment by gingerbenw** [comment]
> Was there a particular reason for placing tests in the same directory as the source files? IMO as we add more files this will make it harder to navigate in a tree view

Old habits die hard, I suppose! it's a pretty typical pattern for React components. I'll move them over to keep the same structure as the other tests.

**Review by yousif-bugsnag — APPROVED** [review]

**Pull request merged**